Fixes #2, Windows support

alexsaveliev opened the pull request:
- fixes #2 by adding basic .PHONY target support
- Make it compile and pass tests on Windows

alexsaveliev committed:
Fixes #2, Windows support

- fixes #2 by adding basic .PHONY target support
- Make it compile and pass tests on Windows

alexsaveliev committed:
Merge remote-tracking branch 'upstream/master'

beyang commented on Makefile:
```diff
@@ -1,4 +1,4 @@
 install: ${GOBIN}/makex
 
-${GOBIN}/makex: $(shell find -type f -and -name '*.go')
+${GOBIN}/makex: $(shell /usr/bin/find -type f -and -name '*.go')
```
Is this necessary for Windows? On some *nix systems, this might be located at `/usr/local/bin/find` or elsewhere.

alexsaveliev replied:
Yes, on Windows in 99% of cases `\Windows\System32\find.exe` will be used (performs text search). I doubt that somebody wants to use `find` from `/usr/local` or other non-standard script/binary from the `PATH` and I think that `/usr/bin` is the correct location everywhere

beyang replied:
Add a check for OS and use `find` for non-Windows and the full path for Windows.

alexsaveliev replied:
> Add a check for OS and use find for non-Windows and the full path for Windows.

Done

beyang commented:
aside from comment, LGTM.

alexsaveliev committed:
Makefile corrections

- using /usr/bin/find on Windows to ensure that Cygwin or MSYS variant is used instead of Windows one; on other platforms using `find`
- added `install` to phony targets
- if there is no GOBIN, using GOPATH/bin as a target folder
